packages/hull-client/src/lib/firehose.js

```
@@ -63,18 +75,13 @@ class FirehoseBatcher {
       sentAt: new Date()
     };
 
-    return this.handler(params, this).then(
+    const flushed = this.handler(params, this);
+    flushed.then(
       ok => fns.forEach(func => func(null, ok)),
       err => fns.forEach(func => func(err, null))
     );
+    return flushed.then(fn, fn);
```

This swallows errors in each `fns`